### ISSUE TITLE
System field and search fields should be handled by the MappingEngine

### ISSUE THREAD

**Issue by manuelbernhardt** [delving/sip-creator#200]
We (Sjoerd and I) just spent quite some time fixing various issues in the hub. One of the conclusions we came to is that the new separation of concerns of having the summaryFields and searchFields processed by the hub, is not very stable and coherent.
- all paths in the record definition are "logical paths", not xpaths, only those two change, so that is kind of incoherent
- it seems that the DOM node tree produced by the mapping engine, when evaluated with XPath, doesn't find results. I have been puzzling around with this now and can't figure out why, but it feels kind of odd in any case. one solution would be to re-parse the serialized XML with a parser I know will work with XPath, but that would be a way too big performance penalty to take, and too much indirection anyhow
- these fields should be part of the unit tests that are part of the record definition development, and the paths checked for validity

Another thing is that the summary fields should, in fact, be called system fields, because that is what they are - they are an interoperability format across various record definitions.

In summary, what we'd need for the hub and the indexing to work is two additional method in the MappingEngine:

```
Map<String, List<String>> getSystemFields()
Map<String, List<String>> getSearchFields()
```

that would be read respectively from the <systemFields> and <searchFields> in the record definition.

Example of such fields are here: https://github.com/delving/definitions/blob/8bad7520fab62fb5dcb8e650cee1012fb4e25351/src/main/resources/definitions/icn/icn-record-definition.xml#L97
